src/commands/lambda/__tests__/flare.test.ts

```diff
@@ -1,21 +1,444 @@
-import {createMockContext, makeCli} from './fixtures'
+import fs from 'fs'
+import process from 'process'
+import * as stream from 'stream'
+import util, {InspectOptions} from 'util'
+
+import axios from 'axios'
+import FormData from 'form-data'
+import JSZip from 'jszip'
+
+import {API_KEY_ENV_VAR, AWS_DEFAULT_REGION_ENV_VAR, CI_API_KEY_ENV_VAR} from '../constants'
+import {LambdaFlareCommand, writeFile, zipContents} from '../flare'
+import {requestAWSCredentials} from '../prompt'
+
+import {createMockContext, makeCli, mockAwsCredentials} from './fixtures'
+
+// Constants
+const MOCK_FOLDER_PATH = './mock_folder'
+const MOCK_FILE_PATH = 'function_config.json'
+const MOCK_ZIP_PATH = 'output.zip'
+const MOCK_API_KEY = 'test-api-key'
+const VALID_INPUT = ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com']
+const MOCK_CONFIG = {
+  Environment: {
+    Variables: {
+      DD_API_KEY: MOCK_API_KEY,
+      DD_SITE: 'datadoghq.com',
+      DD_LOG_LEVEL: 'debug',
+    },
+  },
+  FunctionArn: 'arn:aws:lambda:us-east-1:123456789012:function:some-function',
+  FunctionName: 'some-function',
+}
+
+// Mock util.inspect to remove color codes
+const originalInspect = util.inspect
+jest.spyOn(util, 'inspect').mockImplementation((object: any, options?: InspectOptions) => {
+  return originalInspect(object, {...options, colors: false})
+})
+
+// Commons mocks
+jest.mock('../functions/commons', () => ({
+  getAWSCredentials: jest.fn(),
+  getLambdaFunctionConfig: jest.fn().mockImplementation(() => Promise.resolve(MOCK_CONFIG)),
+  getRegion: jest.requireActual('../functions/commons').getRegion as () => string | undefined,
+}))
+jest.mock('../prompt')
+
+// File system mocks
+jest.mock('fs')
+fs.writeFileSync = jest.fn().mockImplementation(() => {})
+fs.readFileSync = jest.fn().mockReturnValue(JSON.stringify(MOCK_CONFIG, undefined, 2))
+const mockReadStream = new stream.Readable({
+  read() {
+    this.push(JSON.stringify(MOCK_CONFIG, undefined, 2))
+    this.push(undefined)
+  },
+})
+fs.createReadStream = jest.fn().mockReturnValue(mockReadStream)
```

Comment:
Can we have this mocks inside the `jest.mock`? Just like you did for mocking `commons`.

Reply:
I can't because `jest.mock` is ran before all of the imports and consts are defined, even if the imports and consts appear above these lines of code in the file.

An option is to try something like this below, but I still get the error: `Cannot access 'fixtures_1' before initialization` (referring to `mockDatadogApiKey`)

```let MOCK_CONFIG: any
jest.mock('fs', () => {
  MOCK_CONFIG = {
    Environment: {
      Variables: {
        DD_API_KEY: mockDatadogApiKey, // <--- can't be used in jest.mock because it hasn't been initialized
        DD_SITE: 'datadoghq.com',
        DD_LOG_LEVEL: 'debug',
      },
    },
    FunctionArn: 'arn:aws:lambda:us-east-1:123456789012:function:some-function',
    FunctionName: 'some-function',
  }
  const mockReadStream = new stream.Readable({
    read() {
      this.push(JSON.stringify(MOCK_CONFIG, undefined, 2))
      this.push(undefined)
    },
  })

  return {
    writeFileSync: jest.fn().mockImplementation(() => {}),
    readFileSync: jest.fn().mockReturnValue(JSON.stringify(MOCK_CONFIG, undefined, 2)),
    createWriteStream: jest.fn().mockReturnValue(mockReadStream),
  }
})```